torch/onnx/symbolic_opset11.py

```diff
@@ -560,6 +560,26 @@ def mm(g, self, other):
     return g.op("Gemm", self, other, beta_f=0.0, alpha_f=1.0)
 
 
+def index(g, self, index):
+    if sym_help._operator_export_type == torch.onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK:
+        return g.op("ATen", self, index, operator_s="index")
+
+    if sym_help._is_packed_list(index):
+        indices = sym_help._unpack_list(index)
+    else:
+        indices = [index]
+
+    # Handle single mask index.
+    if len(indices) == 1:
+        index = indices[0]
+        if not sym_help._is_none(index) and (index.type().scalarType() == "Bool" or index.type().scalarType() == "Byte"):
+            from torch.onnx.symbolic_opset9 import nonzero
+            index = nonzero(g, index)
+            return g.op('GatherND', self, index)
```

Comment:
Just out of curiosity, ONNX GatherND only takes long dtype for 'index', right?
Does this requirement match the torch 'index' op?

Reply:
yes, torch 'index' requires index type to be long, byte or bool.

On the other hand, this line of code is calling nonzero output as index, and nonzero always outputs long type.